### ISSUE TITLE
New Rule: blank lines around comments [$15]

### ISSUE THREAD

**Issue by nzakas** [eslint/eslint#1344]
This rule should enforce or disallow a blank line before and after comments that appear on their own line. For example, if enforcing a line after a comment, the following would happen:

```js
// some comment

doSomething();
```

This would be an error:

```js
// some comment
doSomething();
```

This would not be an error because the comment isn't on its own line:

```js
doSomething();   // some comment
doSomethingElse();
```

The rule ID should be `lines-around-comment` with options to indicate where blank lines should be enforced. For example:

```json
lines-around-comment: [2, {
    beforeLineComment: false,
    afterLineComment: false,
    beforeBlockComment: false,
    afterBlockComment: true
}
```

**Comment by xjamundx** [comment]
I want this rule. I'll do it.

**Comment by xjamundx** [comment]
Similar JSCS rule for reference:
http://jscs.info/rule/requirePaddingNewLinesBeforeLineComments.html

**Comment by xjamundx** [comment]
@hraban it looks like you started on this a while ago. Need any help?

**Comment by xjamundx** [comment]
I'm thinking the `spaced-line-comment` rule will be a good starting place for working on this...

**Comment by nzakas** [comment]
:+1:

**Comment by hraban** [comment]
Hi guys! I totally forgot about this rule. Feel free to take my code and modify it, or just chuck it and start over.

Good luck!

**Comment by xjamundx** [comment]
A pretty early implementation, with very few tests, but it works for the basic cases so far.
